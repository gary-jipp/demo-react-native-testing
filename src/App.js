import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import HomeStack from 'navigator/HomeStack';

const App = () => {
  return (
    <NavigationContainer>
      <HomeStack />
    </NavigationContainer>
  );
};

export default App;
