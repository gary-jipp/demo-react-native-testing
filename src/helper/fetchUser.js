export default fetchUser = function(email) {
  const url = `https://jsonplaceholder.typicode.com/users?email=${email}`;
  // Call auth provider login API
  return fetch(url)
    .then((response) => response.json())
    .then((res) => {
      return res.length ? res[0] : null;
    });
};