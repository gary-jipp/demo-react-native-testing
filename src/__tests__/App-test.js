import React from 'react';
import {render, cleanup, fireEvent, getQueriesForElement} from "@testing-library/react-native";
import fetchUser from 'helper/fetchUser';
import App from "App";

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');
jest.mock("helper/fetchUser");

// unmounts the components after each test
afterEach(cleanup);

describe("Testing Library Tests for App", () => {

  it("Renders without crashing", () => {
    const {toJSON} = render(<App />);
    // expect(toJSON()).toMatchSnapshot();
  });

  it("Displays Info page after login", async () => {
    // fetchUser will always resolve to mockUser
    fetchUser.mockResolvedValue(mockUser);

    const {toJSON, getByText, findByLabelText, getByPlaceholderText, getByRole, findByText} = render(<App />);

    const username = getByPlaceholderText("email address");
    const password = getByPlaceholderText("password");
    const button = getByRole("button");

    fireEvent.press(button);
    getByText("required", {exact: false});

    fireEvent.changeText(username, 'Sincere@april.biz');
    fireEvent.changeText(password, '123');
    fireEvent.press(button);

    {
      const info = await findByLabelText("Info");
      const {getByText, getByRole, toJSON} = getQueriesForElement(info);
      getByText("Leanne Graham");

      const button = getByRole("button");
      fireEvent.press(button);
      getByText('harness real-time e-markets');
    }
  });

});

const mockUser =
{
  "id": 1,
  "name": "Leanne Graham",
  "username": "Bret",
  "email": "Sincere@april.biz",
  "address": {
    "street": "Kulas Light",
    "suite": "Apt. 556",
    "city": "Gwenborough",
    "zipcode": "92998-3874",
    "geo": {
      "lat": "-37.3159",
      "lng": "81.1496"
    }
  },
  "phone": "1-770-736-8031 x56442",
  "website": "hildegard.org",
  "company": {
    "name": "Romaguera-Crona",
    "catchPhrase": "Multi-layered client-server neural-net",
    "bs": "harness real-time e-markets"
  }
};
