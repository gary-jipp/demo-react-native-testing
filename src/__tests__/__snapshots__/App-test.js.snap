// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Library Tests for App Renders without crashing 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNSScreenStack
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      stackPresentation="push"
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    >
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="Sign In"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            Object {
              "backgroundColor": "rgb(242, 242, 242)",
            },
            undefined,
          ]
        }
      >
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "alignItems": "stretch",
              "backgroundColor": "#fff",
              "flex": 1,
              "paddingHorizontal": 40,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#353031",
                "fontSize": 34,
                "fontWeight": "bold",
                "marginBottom": 10,
                "marginTop": 10,
              }
            }
          >
            Login
          </Text>
          <View
            style={
              Object {
                "alignItems": "stretch",
                "backgroundColor": "#fff",
                "flex": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#f4f6f8",
                    "borderColor": "#f4f6f8",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "marginVertical": 5,
                    "paddingHorizontal": 15,
                    "paddingVertical": 5,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <TextInput
                  autoCapitalize="none"
                  onChangeText={[Function]}
                  placeholder="email address"
                  style={
                    Object {
                      "color": "#353031",
                      "flex": 1,
                      "fontSize": 24,
                      "marginRight": 10,
                      "marginTop": 3,
                    }
                  }
                  testID="SignIn.usernameInput"
                />
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "#f4f6f8",
                    "borderColor": "#f4f6f8",
                    "borderRadius": 10,
                    "borderWidth": 1,
                    "marginVertical": 5,
                    "paddingHorizontal": 15,
                    "paddingVertical": 5,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                    "justifyContent": "space-between",
                  }
                }
              >
                <TextInput
                  autoCapitalize="none"
                  onChangeText={[Function]}
                  placeholder="password"
                  secureTextEntry={true}
                  style={
                    Object {
                      "color": "#353031",
                      "flex": 1,
                      "fontSize": 24,
                      "marginRight": 10,
                      "marginTop": 3,
                    }
                  }
                  testID="SignIn.passwordInput"
                />
              </View>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": false,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="button"
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "#2196F3",
                      "borderRadius": 5,
                      "elevation": 0,
                      "justifyContent": "center",
                      "paddingHorizontal": 20,
                      "paddingVertical": 12,
                    },
                    Object {
                      "alignItems": "center",
                      "borderRadius": 10,
                      "justifyContent": "center",
                      "marginTop": 10,
                      "paddingVertical": 16,
                    },
                    false,
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#EEE",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "letterSpacing": 0.25,
                        "lineHeight": 21,
                      },
                      Object {
                        "color": "#fff",
                        "fontSize": 20,
                        "fontWeight": "600",
                      },
                    ]
                  }
                >
                  Login
                </Text>
              </View>
            </View>
          </View>
        </RCTSafeAreaView>
      </View>
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
